tests/unit/lms/services/canvas_api/_basic_test.py

```diff
@@ -79,19 +79,27 @@ def test_send_raises_CanvasAPIError_for_error_responses(
     ):
         http_session.send.return_value = factories.requests.Response(status_code=501)
 
-        with pytest.raises(CanvasAPIError):
+        with pytest.raises(CanvasAPIError) as exc_info:
             basic_client.send("METHOD", "path/", schema=Schema)
 
+        exc = exc_info.value
+        assert exc.request == http_session.send.call_args[0][0]
```

Comment:
This is true and compact, butI found it a bit hard to grok at first pass. Not sure if something like this might help?

```suggestion
        assert exc.request == http_session.send.call_args[0][0]  # First positional arg is request
```